activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -242,7 +242,16 @@ def references!(*table_names) # :nodoc:
     #   Model.select(:field).first.other_field
     #   # => ActiveModel::MissingAttributeError: missing attribute: other_field
     def select(*fields)
-      return super if block_given?
+      if block_given?
+        if fields
+          ActiveSupport::Deprecation.warn(<<-WARNING.squish)
```

Comment:
This deprecation affects all relations. I think it is better to move it in collection proxy.